fix(curriculum): clarify that replace() returns a new string in JS lesson

NiloySaha84 opened the pull request:
…bility

Checklist:

<!-- Please follow this checklist and put an x in each of the boxes, like this: [x]. It will ensure that our team takes your pull request seriously. -->

- [x] I have read and followed the [contribution guidelines](https://contribute.freecodecamp.org).
- [x] I have read and followed the [how to open a pull request guide](https://contribute.freecodecamp.org/how-to-open-a-pull-request/).
- [x] My pull request targets the `main` branch of freeCodeCamp.
- [x] I have tested these changes either locally on my machine, or Gitpod.

<!--If your pull request closes a GitHub issue, replace the XXXXX below with the issue number.-->

Closes #61484 

<!-- Feel free to add any additional description of changes below this line -->

NiloySaha84 committed:
feat: feat: clarify that replace() returns a new string due to immutability

huyenltnguyen reviewed (commented):
Please help apply the change to the following files as well:
- https://github.com/freeCodeCamp/freeCodeCamp/blob/91f87ea237a23f6c1eb8492cd4b5976fe3b458aa/curriculum/challenges/english/25-front-end-development/review-javascript/6723d3cfdd0717d3f1bf27e3.md#L303
- https://github.com/freeCodeCamp/freeCodeCamp/blob/91f87ea237a23f6c1eb8492cd4b5976fe3b458aa/curriculum/challenges/english/25-front-end-development/review-javascript-strings/6723c1946e4cd7909a836bb4.md#L101

NiloySaha84 committed:
Update curriculum/challenges/english/25-front-end-development/lecture-working-with-common-string-methods/67326c29dcd98fc5ecc49779.md

Co-authored-by: Huyen Nguyen <25715018+huyenltnguyen@users.noreply.github.com>

majestic-owl448 committed:
Merge branch 'main' of https://github.com/freeCodeCamp/freeCodeCamp into pr/NiloySaha84/61730

ilenia-magoni commented:
I have reseolved the merge conflict

NiloySaha84 committed:
WIP: partial edits before merging

NiloySaha84 commented:
> > I have done it and submitted a pull request naming "fix: fixed the replace method in review-javascript and review-javascr… #61816"
> 
> Please help add those changes to this PR.

done

> > I have done it and submitted a pull request naming "fix: fixed the replace method in review-javascript and review-javascr… #61816"
> 
> Please help add those changes to this PR.

done
